### PR TITLE
Upgrade nix to v0.28.0

### PR DESCRIPTION
**Pull request by charliermarsh**
See: https://github.com/watchexec/clearscreen/pull/20.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -759,14 +758,14 @@ mod unix {
...
 
 	fn write_termios(f: impl Fn(&mut Termios)) -> Result<(), Error> {
-		if isatty(STDIN_FILENO)? {
-			let mut t = tcgetattr(STDIN_FILENO)?;
+		if isatty(stdin().as_raw_fd())? {
+			let mut t = tcgetattr(stdin().as_fd())?;
 			reset_termios(&mut t);
 			f(&mut t);
-			tcsetattr(STDIN_FILENO, TCSANOW, &t)?;
+			tcsetattr(stdin().as_fd(), TCSANOW, &t)?;
```
Comment (charliermarsh) [inline src/lib.rs:765]: I'm not certain if this is the correct way to do this, but it was the easiest way I could find that didn't involve unsafe.

Comment (charliermarsh) [reply src/lib.rs:765]: (These methods changed to requiring `AsFd` rather than `AsRawFd`.)

**Comment by passcod** [comment]
Thank you! I'll get this released ASAP

**Comment by charliermarsh** [comment]
Thank you :)

I'm flying slightly blind on the `AsFd` thing so let me know if that looks off.

**Comment by passcod** [comment]
Done in 3.0.0

**Pull request merged**